### ISSUE TITLE
Tabs control does not work with screen readers

### ISSUE THREAD

**Issue by carmacleod** [carbon-design-system/carbon#6984]
The Tabs control doesn't work with JAWS or NVDA screen readers. They can sort of be used in reading/browse mode, but they don't recognize the ARIA tabs in focus mode, so they eat the arrow keys, which is not supposed to happen (i.e. if a screen reader recognizes a tablist with tab children, they will pass the arrow keys through to the control so that the user can switch tabs).
Note: Only tried on Windows so far, but I expect to see a similar problem on Mac. Also, didn't try vanilla Carbon Tabs - only react.

The problem is that the "tab" is not correctly marked up. The "tab" role needs to be on "the element that takes focus and does the key event handling". This is true of all interactive ARIA roles - the role needs to be on the element that the user will actually interact with.

In the case of the Carbon Tabs control, the interaction is taking place on the anchor element, but the list item element has all of the ARIA markup and tabindex="-1" as well.

i.e. a Carbon Tab currently looks like this:
```
<li role="tab" tabindex="-1" class="bx--tabs--scrollable__nav-item" aria-selected="false" aria-disabled="false" aria-controls="tab-1__panel" wiid="C11">
    <a id="tab-1" class="bx--tabs--scrollable__nav-link" href="#" tabindex="-1" wiid="C12">Tab label 1</a>
</li>
```

but it needs to _either_ look like the following (i.e. move all the things over to the anchor, and make the li presentational so that screen readers don't get confused by the li's implicit listitem role):
```
<li role="presentation" class="bx--tabs--scrollable__nav-item" wiid="C11">
    <a id="tab-1" role="tab" class="bx--tabs--scrollable__nav-link" href="#" tabindex="-1" aria-selected="false" aria-disabled="false" aria-controls="tab-1__panel" wiid="C12">Tab label 1</a>
</li>
```

OR, alternatively, a Carbon Tab can look like this (i.e. use a button instead of an anchor, so that the context menu for the tab doesn't have linky things in it like "Open link in new window" or "Copy link address"):
```
<li role="presentation" class="bx--tabs--scrollable__nav-item" wiid="C11">
    <button id="tab-1" role="tab" class="bx--tabs--scrollable__nav-link" tabindex="-1" aria-selected="false" aria-disabled="false" aria-controls="tab-1__panel" wiid="C12">Tab label 1</button>
</li>
```

OR, another alternative is to go completely fake and use divs (which is what the [APG](https://w3c.github.io/aria-practices/examples/tabs/tabs-1/tabs.html) did), which would be ok, too, but I'm not sure if that's as nice for "progressive enhancement" (i.e. no js, older browsers, etc).

Happy to work with you to get this working properly.
